component/src/org/wso2/ballerina/connectors/oauth2/ClientConnector.bal

```diff
@@ -0,0 +1,172 @@
+
+        return response, e;
+    }
+
+    @Description { value:"Put with OAuth2 authentication"}
+    @Param { value:"path: The endpoint path"}
+    @Param { value:"request: The request of the method"}
+    @Return { value:"response object"}
+    @Return { value:"Error occured during HTTP client invocation." }
+    action put (string path, http:Request originalRequest) (http:Response, http:HttpConnectorError) {
+        json originalPayload = originalRequest.getJsonPayload();
+
+        populateAuthHeader(originalRequest, accessToken);
+        response, e = httpConnectorEP.put (path, originalRequest);
+
+        http:Request request = {};
+        request.setJsonPayload(originalPayload);
+
+        if (checkAndRefreshToken(request, accessToken, clientId, clientSecret, refreshToken, refreshTokenEP,
+                                 refreshTokenPath)) {
+            response, e = httpConnectorEP.put (path, request);
+        }
+
+        return response, e;
+    }
+
+    @Description { value:"Delete with OAuth2 authentication"}
+    @Param { value:"path: The endpoint path"}
+    @Param { value:"request: The request of the method"}
+    @Return { value:"response object"}
+    @Return { value:"Error occured during HTTP client invocation." }
+    action delete (string path, http:Request originalRequest) (http:Response, http:HttpConnectorError) {
+        json originalPayload = originalRequest.getJsonPayload();
+
+        populateAuthHeader(originalRequest, accessToken);
+        response, e = httpConnectorEP.delete (path, originalRequest);
+
+        http:Request request = {};
+        request.setJsonPayload(originalPayload);
+
+        if (checkAndRefreshToken(request, accessToken, clientId, clientSecret, refreshToken, refreshTokenEP,
+                                 refreshTokenPath)) {
+            response, e = httpConnectorEP.delete (path, request);
+        }
+
+        return response, e;
+    }
+
+    @Description { value:"Patch with OAuth2 authentication"}
+    @Param { value:"path: The endpoint path"}
+    @Param { value:"request: The request of the method"}
+    @Return { value:"response object"}
+    @Return { value:"Error occured during HTTP client invocation." }
+    action patch (string path, http:Request originalRequest) (http:Response, http:HttpConnectorError) {
+        json originalPayload = originalRequest.getJsonPayload();
+
+        populateAuthHeader(originalRequest, accessToken);
+        response, e = httpConnectorEP.patch (path, originalRequest);
+
+        http:Request request = {};
+        request.setJsonPayload(originalPayload);
+
+        if (checkAndRefreshToken(request, accessToken, clientId, clientSecret, refreshToken, refreshTokenEP,
+                                 refreshTokenPath)) {
+            response, e = httpConnectorEP.patch (path, request);
+        }
+
+        return response, e;
+    }
+}
+
+function populateAuthHeader (http:Request request, string accessToken) {
+    if (accessTokenValue == "") {
```

Comment:
If you put "|| accessTokenValue == null", this code will also work with Ballerina 0.95.3, since from there onwards the default value of string would be null.

Reply:
On second thought, let's just make it work from 0.95.3 onwards. So basically just check for == null.